src/clj/ataru/applications/application_service.clj

```diff
@@ -337,9 +338,18 @@
                                        "created-time"   {:key          (:key a)
                                                          :created-time (:created-time a)})}))}))
 
+(defn- hakukohteiden-ehdolliset
+  [valinta-tulos-service applications]
+  (->> (mapcat :hakukohde applications)
```

Comment:
Oon jo pitkään kelannut että voitaisiin ottaa `net.cgrand/xforms` -riippuvuus vihdoinkin projektiin koska se on hyödyllinen. Jos ottaisi sen nyt tässä branchissa riippuvuudeksi:

`[net.cgrand/xforms "0.19.2"]`

niin tämän funktion voi kirjoittaa uudelleen mm. näin:

```
(ns ataru.applications.application-service
  (:require
    ..
    [net.cgrand.xforms :as x]))

(x/into {}
        (comp (mapcat :hakukohde)
              (dedupe)
              (map #(vector % (valinta-tulos-service/hakukohteen-ehdolliset
                                valinta-tulos-service
                                %))))
        applications)
```

Ilman transduceria nykyinen toteutus suoriutuu mielivaltaisen datan käsittelystä suunnilleen seuraavasti:

             Execution time mean : 1,099397 ms

Tuolla transducer-toteutuksella samasta datasta suoriudutaan suunnilleen seuraavasti:

             Execution time mean : 255,146261 µs